File: 1468-check-if-n-and-its-double-exist/check-if-n-and-its-double-exist.py
class Solution:
    def checkIfExist(self, arr: List[int]) -> bool:
        visited = set()

        for i in arr:
            if 2 * i in visited or (i % 2 == 0 and i // 2 in visited):
                return True
            visited.add(i)


        return False

File: 1468-check-if-n-and-its-double-exist/check-if-n-and-its-double-exist.java
class Solution {
    public boolean checkIfExist(int[] arr) {
        HashSet<Integer> visited = new HashSet<Integer>();

        for (int i: arr){
            if (visited.contains(i * 2) || (i  % 2 == 0 && visited.contains(i / 2)))
                return true;
            visited.add(i);
        }
        return false; 
    }
}